src/loginAzurePowerShell.ts

```diff
@@ -0,0 +1,76 @@
+import * as os from 'os';
+import * as core from '@actions/core';
+import * as exec from '@actions/exec';
+import * as io from '@actions/io';
+
+var psPath: string;
+
+export const initializeAz = async (servicePrincipalId: string, servicePrincipalKey: string, tenantId: string, subscriptionId: string) => {
+    psPath = await io.which("pwsh", true);
+    await importModule();
+    await loginToAzure(servicePrincipalId, servicePrincipalKey, tenantId, subscriptionId);
+}
+
+async function importModule() {
+    setPSModulePath();
+    const prefix = "az_";
+    const moduleName: string = "Az.Accounts";
+    const azLatestVersion: string = await getLatestModule(moduleName);
+    core.debug(`Az Module version used: ${azLatestVersion}`);
+    setPSModulePath(`${prefix}${azLatestVersion}`);
+}
+
+function setPSModulePath(azPSVersion: string = "") {
```

Comment:
add function to validate azPSVersion is a well formed semantic version (<Major>.<Minor>.<Patch>)